ec/ecresource/deploymentresource/elasticsearch_expanders.go

```diff
@@ -130,6 +131,38 @@ func expandEsTopology(raw interface{}, topologies []*models.ElasticsearchCluster
 			}
 		}
 
+		if ntData, ok := topology["node_type_data"]; ok && ntData.(string) != "" {
+			nt, err := strconv.ParseBool(ntData.(string))
+			if err != nil {
+				return nil, err
```

Comment:
Can we wrap this error with a meaningful string? I think it could end up being very confusing for users to receive the error that `strong.ParseBool` returns as is.

```suggestion
				return nil, fmt.Errorf("failed parsing node_type_data value: %w", err)
```